packages/common/src/hooks/useDialog/useDialog.test.tsx

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import { act, render, waitFor } from '@testing-library/react';
+import { TestingProvider } from '@openmsupply-client/common';
+
+import { useDialog } from './useDialog';
+
+describe('Footer', () => {
+  const DialogExample: React.FC = () => {
+    const { Modal, showDialog } = useDialog({
+      body: <div>dialog body context</div>,
+    });
+
+    return (
+      <div>
+        {Modal}
+        <button onClick={showDialog}>show dialog</button>
+      </div>
+    );
+  };
+
+  it('Dialog not shown when first rendered', () => {
+    const { queryByText } = render(
+      <TestingProvider>
+        <DialogExample />
+      </TestingProvider>
+    );
+
+    expect(queryByText(/dialog body context/)).not.toBeInTheDocument();
+  });
+
+  it('Dialog is shown when requested', () => {
+    const { getByRole, getByText } = render(
+      <TestingProvider>
+        <DialogExample />
+      </TestingProvider>
+    );
+
+    act(() => getByRole('button', { name: 'show dialog' }).click());
+
+    expect(getByText(/dialog body context/)).toBeInTheDocument();
+  });
+
+  it('Cancel button is shown', () => {
+    const { getByRole } = render(
+      <TestingProvider>
+        <DialogExample />
+      </TestingProvider>
+    );
+
+    act(() => getByRole('button', { name: 'show dialog' }).click());
+
+    expect(getByRole('button', { name: 'Cancel' })).toBeInTheDocument();
+  });
+
+  it('Dialog is hidden when cancelled', async () => {
+    const { getByRole, queryByText } = render(
+      <TestingProvider>
+        <DialogExample />
+      </TestingProvider>
+    );
+
+    act(() => getByRole('button', { name: 'show dialog' }).click());
+
+    act(() => getByRole('button', { name: 'Cancel' }).click());
+
+    await waitFor(() => {
+      expect(queryByText(/dialog body context/)).not.toBeInTheDocument();
+    });
```

Comment:
general nit about using `/regex/i` to ignore the casing

Reply:
sure. changed - I didn't for this one because the text is specified above in this test file. using the case makes it more strict and therefore better, right?